### ISSUE TITLE
Add accessibility support for individuals with disabilities

### ISSUE THREAD

**Issue by ethindp** [novnc/noVNC#1877]
**Describe the bug**
Currently, the NoVNC web console is not accessible to screen readers and other assistive technologies. This makes it extremely difficult (if not pretty much impossible) to use.

**To Reproduce**
1. Start any screen reader of your choosing
2. Try to log into any NoVNC console or VPS provider that uses it
3. Notice that you will neither be able to read the input window nor anything that the system outputs without OCR technology or other workarounds. OCR is notoriously unreliable and depends on far too many factors to guarantee such, even if AI helps.

**Expected behavior**
It should be possible for individuals who use assistive technologies, such as screen readers, to use the NoVNC console, as this is one thing many (if not all) VPS providers universally use, and being unable to locks them out from many providers who do not provide an alternative.

I classify this as a bug, not a feature, because classifying it as a feature implies that it's optional when it shouldn't be. I'm uncertain if the lack of accessibility is due to failures or problems in the actual VNC protocol itself, or just in the clients, so I thought that, given how widespread this particular client is, it would be worth discussing this problem to see what is achievable and what is not. If this is a problem with the underlying protocol, I'm not sure where to go from here, though my (admittedly naive understanding of this particular protocol and setup) makes me think it should (theoretically) be possible.

**Comment by CendioOssman** [comment]
I'm afraid VNC only transfers an image, so there is no information available that we could provide to screen readers. So yes, it's a protocol issue.

That unfortunately means that it is not something noVNC can do much about on its own. You'd need to get a server on board to provide the necessary protocol extensions.

And such a protocol would likely require quite a bit of work. VNC is a cross-platform system, which means a protocol would need to be designed in a way that can bridge at least Wayland, X11, macOS and Windows.

Given that no one is looking into this, I'm afraid I don't see much value in having an issue open in our tracker for this. We can reopen this if a VNC server starts looking at adding this functionality. But right now, it's outside the scope of what the noVNC project can do.
